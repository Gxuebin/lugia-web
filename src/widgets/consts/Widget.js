/**
 *
 * create by ligx
 *
 * @flow
 */
export const Input = 'sv.widget.Input';
export const Align = 'sv.widget.Align';
export const Trigger = 'sv.widget.Trigger';
export const InputTag = 'sv.widget.InputTag';
export const Theme = 'sv.widget.Theme';
export const Menu = 'sv.widget.Menu';
export const DropMenu = 'sv.widget.DropMenu';
export const Tree = 'sv.widget.Tree';
export const ThrottleScroller = 'sv.widget.ThrottleScroller';
