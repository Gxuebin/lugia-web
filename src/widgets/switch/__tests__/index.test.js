/**
 *
 * create by ZhangBoPing
 *
 * @flow
 */
import React from 'react';
import chai from 'chai';

import 'jest-styled-components';
import Enzyme, { mount, shallow, } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import Switch from '../';
import Switch from '../switch';
import renderer from 'react-test-renderer';
import Widget from '../../consts/index';

const { expect: exp, } = chai;

Enzyme.configure({ adapter: new Adapter(), });

describe('Switch', () => {
  it('snapshot', () => {
    const Target = <Switch />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();

  });

  it('API: defaultChecked', () => {
    const Wrapper = shallow(<Switch defaultChecked />);
    exp(Wrapper.state('checked')).to.eql(true);
  });

  it('API: checked', () => {
    const Wrapper = shallow(<Switch checked />);
    exp(Wrapper.state('checked')).to.eql(true);
    Wrapper.setState({checked: false,});
    exp(Wrapper.state('checked')).to.eql(false);
  });

  it('API: disabled', () => {
    const Wrapper = shallow(<Switch disabled />);
    exp(Wrapper.state('disabled')).to.eql(true);
    Wrapper.setState({disabled: false,});
    exp(Wrapper.state('disabled')).to.eql(false);
  });

  it('API: checkedChildren', () => {
    const Wrapper = shallow(<Switch checkedChildren={'on'} />);
    Wrapper.setState({checked: true,});
    exp(Wrapper.html().indexOf('on') > -1).to.eql(true);
  });

  it('API: unCheckedChildren', () => {
    const Wrapper = shallow(<Switch unCheckedChildren={'off'} />);
    Wrapper.setState({checked: false,});
    exp(Wrapper.html().indexOf('off') > -1).to.eql(true);
  });

  /* has problem => autoFocus is not work. */
  it('API: autoFocus', () => {
    const Wrapper = mount(<Switch autoFocus />);
    const checkedState = Wrapper.state('checked');
    Wrapper.find('span').at(0).simulate('keyDown', {keyCode: 13,});
    exp(Wrapper.state('checked')).to.eql(!checkedState);
  });
  
  // it('normal type: danger', () => {
  //   const Target = <Button type="danger">hello</Button>;
  //   expect(renderer.create(Target).toJSON()).toMatchSnapshot();

  // });
  // it('props: click', async () => {
  //   let onClick;
  //   const promise = new Promise(resolve => {
  //     onClick = e => {
  //       resolve(e.target);
  //     };
  //   });
  //   const cmp = mount(<Button type="danger" onClick={onClick}>hello</Button>);
  //   let target = { px: 'hello', };
  //   cmp.find('hello').at(0).simulate('click', { target, });

  //   exp(await  promise).to.be.eql(target);
  // });

});
