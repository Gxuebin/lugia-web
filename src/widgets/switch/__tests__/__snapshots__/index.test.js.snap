// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch snapshot 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  height: 22px;
  min-width: 44px;
  line-height: 20px;
  vertical-align: middle;
  border-radius: 100px;
  border: 1px solid transparent;
  background-color: rgba(0,0,0,.25);
  cursor: 'pointer';
  -webkit-transition: all .36s;
  transition: all .36s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:before,
.c0:after {
  position: absolute;
  width: 18px;
  height: 18px;
  left: 1px;
  top: 1px;
  border-radius: 18px;
  background-color: #fff;
  content: " ";
  cursor: pointer;
  -webkit-transition: all .36s cubic-bezier(0.78,0.14,0.15,0.86);
  transition: all .36s cubic-bezier(0.78,0.14,0.15,0.86);
}

.c0:active:before,
.c0:active:after {
  width: 24px;
}

.c0:after {
  box-shadow: 0 2px 4px 0 rgba(0,35,11,.2);
}

.c0:before {
  content: "e64d";
  font-family: anticon;
  -webkit-animation: loadingCircle 1s infinite linear;
  animation: loadingCircle 1s infinite linear;
  text-align: center;
  background: transparent;
  z-index: 1;
  display: none;
  font-size: 12px;
}

.c0:focus {
  box-shadow: 0 0 0 2px rgba(24,144,255,.2);
  outline: 0;
}

.c0:focus:hover {
  box-shadow: none;
}

.c1 {
  color: #fff;
  font-size: 12px;
  margin-left: 24px;
  margin-right: 6px;
  display: block;
}

<span
  className="c0"
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <span
    className="c1"
  >
    
  </span>
</span>
`;
